### ISSUE TITLE
SentrySdk.CaptureUserFeedback() not working correctly with Sentry web app

### ISSUE THREAD

**Issue by cyber960** [getsentry/sentry-dotnet#839]
When I use SentrySdk.CaptureUserFeedback in a sample .NET 4.6.1 app with v3.0.6 of sentry-dotnet, the code runs fine, but the Sentry web app does not display the issue and instead shows an error. If I dont have the call to SentrySdk.CaptureUserFeedback and just simply have the call to SentrySdk.CaptureException everything works as expected and the Sentry web app displays the issue as expected.

    const string Dsn = "https://abc@xyz.ingest.sentry.io/1234567"; //insert your own Dsn here
    using (SentrySdk.Init(Dsn))
    {
        try
        {
            int num = 0;
            int val = 6 / num;
        }
        catch (Exception ex)
        {
            SentrySdk.CaptureException(ex);
            SentrySdk.CaptureUserFeedback(new UserFeedback(SentrySdk.LastEventId, "Test Name", "test@gmail.com", "test comments"));
        }
    }

![image](https://user-images.githubusercontent.com/78977691/108583680-51d8e180-7309-11eb-992b-3680fcc37241.png)

**Comment by bruno-garcia** [comment]
The Id is returned by Capture, can you can pass that Id in instead?

https://docs.sentry.io/platforms/dotnet/enriching-events/user-feedback/#use-the-net-sdk

```
var eventId = SentrySdk.CaptureMessage("An event that will receive user feedback.");
SentrySdk.CaptureUserFeedback(eventId, "user@example.com", "It broke.", "The User");
```


Reading `LastEventId` I'm surprised your snippet didn't work either, so we might have a problem with `LastEventId` actually:
https://github.com/getsentry/sentry-dotnet/blob/cee8cc337cec426373c44521c4421e93d4b5c108/src/Sentry/Internal/Hub.cs#L188

**Comment by cyber960** [comment]
Thanks for your reply. I didn't know about the return value, yes it makes more sense to use it instead of LastEventId.

The problem still occurs with the return value and sometimes (maybe 10%?) it works correctly. If I add a Thread.Sleep(500) call before calling CaptureUserFeedback it seems to be working correctly consistently. Maybe there is a syncronization issue happening where the web app is processing user feedback before fully processing the call to capture message?

**Comment by bruno-garcia** [comment]
That's surprising because the the code is synchronous to the point of queueing the event to Sentry but assigning the event id to `LastEventId` is done before returning from `CaptureEvent` so it should always be there. Doesn't mean the event will be sent to Sentry before the user feedback is created and queued for submission but that doesn't matter. Sentry also support receiving them in reversed order.

I also tried this locally and I couldn't reproduce it. The `SentrySdk.LastEventId` property had the event id correctly set. Is this happening in high volume where possible the event queue is full? The Sentry SDK can send a lot of events per second but if you queue more than it can process, it starts dropping them, and `LastEventId` would return `00000..`

**Comment by cyber960** [comment]
This isn't happening at high volume. Just a single event in a test app.

Do you work at Sentry? Would it be helpful to know the issue id (from the URL) so that it could be investigated on the web app side? I am doing this in a new test account at the moment. I can also click the "Fill out a report" button on the web app and make a comment that links back to this issue.

**Comment by bruno-garcia** [comment]
@cyber960 would be nice to have a reproducible project we can run because if I copied the two lines you had on your screenshot and run on a console application, it works as intended.

Did you check the docs for ASP.NET or ASP.NET Core?

https://docs.sentry.io/platforms/dotnet/guides/aspnetcore/enriching-events/user-feedback/#on-the-web

https://docs.sentry.io/platforms/dotnet/guides/aspnet/enriching-events/user-feedback/#on-the-web

These include the snippet to show the user feedback for the user on the website.

Did you try it as documented using the SentryId returned from the CaptureException methog?

**Comment by cyber960** [comment]
yes, I used the SentryId returned from the call to CaptureException.

The problem with viewing the user feedback seems to be intermittent for me, sometimes it works and other times it doesn't. One thing I noticed today is that the error message saying events could not be found shown on the Sentry website seems to disappear and fix itself 20-30 minutes after the issue is sent. I hadn't noticed this before as I was deleting the issues shortly after they were created.
